### PR TITLE
CORE-45716 - sendEvent command promise is resolved with undefined

### PR DESCRIPTION
**Pull request by ninaceban**
Fixing the result of `sendEvent` command promise

## Description

Fixing the result of `sendEvent` command promise

## Related Issue

NONE

## Motivation and Context

Fixing the result of `sendEvent` command promise

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

**Comment by Aaronius** [comment]
It should already be handled systemically. @ninaceban, can you help figure out why this code isn't causing an object to return instead of `undefined`: https://github.com/adobe/alloy/blob/610f1203b5f2c8bc0926c2127898f9869c105608/src/core/executeCommandFactory.js#L99-L100

File: src/components/Identity/createComponent.js [hunk 1]
```diff
@@ -35,7 +36,7 @@ export default ({
           }
         }
 
-        return handleResponseForIdSyncs(response);
+        return handleResponseForIdSyncs(response).then(noop);
```
Comment (Aaronius) [inline src/components/Identity/createComponent.js:39]: @ninaceban I think I understand the problem now. It seems that `processIdSyncs` is resolving the returned promise with an array of `undefined` values. I think the proper thing to do here would be to move the `.then(noop)` to inside `processIdSyncsFactory`. The reasoning is that an array of `undefined` values isn't useful outside that module, so we should be keep it from leaking out of that module. If I'm understanding this right.

Comment (ninaceban) [reply src/components/Identity/createComponent.js:39]: yes, thats right, that promise gets resolved with an array of `undefined`. I'll change `then(noop)` to `processIdSyncsFactory`. Thank you

File: test/unit/specs/components/Identity/createComponent.spec.js [hunk 2]
```diff
@@ -111,7 +115,7 @@ describe("Identity::createComponent", () => {
     const response = { type: "response" };
     const result = component.lifecycle.onResponse({ response });
     expect(handleResponseForIdSyncs).toHaveBeenCalledWith(response);
-    expect(result).toBe(idSyncsPromise);
+    expectAsync(result).toBeResolvedTo(response);
```
Comment (Aaronius) [inline test/unit/specs/components/Identity/createComponent.spec.js:118]: I think you have to do `return expectAsync`

Comment (ninaceban) [reply test/unit/specs/components/Identity/createComponent.spec.js]: thank you!

**Pull request merged**